app/src/main/java/org/simple/clinic/textInputdatepicker/TextInputDatePickerValidator.kt

```diff
@@ -0,0 +1,38 @@
+package org.simple.clinic.textInputdatepicker
+
+import org.simple.clinic.textInputdatepicker.TextInputDatePickerValidator.Result.Notvalid.DateIsInPast
+import org.simple.clinic.textInputdatepicker.TextInputDatePickerValidator.Result.Notvalid.InvalidPattern
+import org.simple.clinic.textInputdatepicker.TextInputDatePickerValidator.Result.Notvalid.MaximumAllowedDateRange
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.time.format.DateTimeParseException
+import javax.inject.Inject
+import javax.inject.Named
+
+class TextInputDatePickerValidator @Inject constructor(
+    @Named("exact_date") private val userEnteredDateFormat: DateTimeFormatter
```

Comment:
Let's use `date_for_user_input`, one of the reasons is the dateformatter that we get with `exact_date` is dependent on the `Country`. So we decided to show this date picker for some other `Country` the date format will not match.